src/tabs/tabs.tsx

```diff
@@ -143,33 +151,112 @@ type TabListProps = (
      * Controls the spacing between tabs
      */
     space?: Space
-}
+
+    /**
+     * The width of the tab list.
+     *
+     * - `'maxContent'`: Each tab will be as wide as its content.
+     * - `'full'`: Each tab will be as wide as the tab list.
+     *
+     * @default 'maxContent'
+     */
+    width?: 'maxContent' | 'full'
+
+    /**
+     * How to align the tabs within the tab list.
+     *
+     * @default 'start'
+     */
+    align?: ResponsiveProp<'start' | 'center' | 'end'>
+} & ObfuscatedClassName
 
 /**
  * A component used to group `<Tab>` elements together.
  */
-function TabList({ children, space, ...props }: TabListProps): React.ReactElement | null {
+function TabList({
+    children,
+    space,
+    width = 'maxContent',
+    align = 'start',
+    exceptionallySetClassName,
+    ...props
+}: TabListProps): React.ReactElement | null {
     const tabContextValue = React.useContext(TabsContext)
 
+    const [selectedTabElement, setSelectedTabElement] = React.useState<HTMLElement | null>(null)
+    const [selectedTabStyle, setSelectedTabStyle] = React.useState<React.CSSProperties>({})
+    const tabListRef = React.useRef<HTMLDivElement>(null)
+
+    const selectedId = tabContextValue?.tabStore.useState('selectedId')
+
+    React.useLayoutEffect(() => {
+        if (!selectedId || !tabListRef.current) {
+            return
+        }
+
+        const tabs = tabListRef.current.querySelectorAll('[role="tab"]')
+
+        const selectedTab = Array.from(tabs).find(
+            (tab) => tab.getAttribute('id') === selectedId,
+        ) as HTMLElement | undefined
+
+        if (selectedTab) {
+            setSelectedTabElement(selectedTab)
+
+            setSelectedTabStyle({
+                left: `${selectedTab.offsetLeft}px`,
+                width: `${selectedTab.offsetWidth}px`,
+            })
+        }
+    }, [selectedId])
+
     if (!tabContextValue) {
         return null
     }
 
     const { tabStore, variant } = tabContextValue
 
     return (
-        // The extra <div> prevents <Inline>'s negative margins from collapsing when used in a flex container
-        // which will render the track with the wrong height
-        <div>
+        // This extra <Box> not only provides alignment for the tabs, but also prevents <Inline>'s
+        // negative margins from collapsing when used in a flex container which will render the
+        // track with the wrong height
+        <Box
+            display="flex"
+            justifyContent={
+                width === 'full'
```

Comment:
Might be easier to see the relationship between align's and justifyContent's values if we constructed a hash like this:

```ts
const justifyContentAlignMap = { 
    start: 'flexStart',
    end: 'flexEnd',
    center: 'center'
}

justifyContent={width === 'full' ? 'center' : justfiyContentAlignMap[align]}
```

Reply:
Sounds good, I'll make the change.